test/Exceptions_test.cpp

```diff
@@ -80,3 +80,15 @@ TEST(Exceptions, checkpoints)
     EXPECT_NO_THROW(t.incrementAdjoint(x.getSlot(), 1.0));
     EXPECT_THROW(t.incrementAdjoint(12312, 1.0), xad::OutOfRange);
 }
+
+TEST(Exceptions, noTape)
+{
+    xad::AD x = 1.0;
+    EXPECT_THROW(x.setDerivative(1.0), xad::NoTapeException);
+    EXPECT_THROW(derivative(x) = 1.0, xad::NoTapeException);
+    EXPECT_THROW(x.derivative(), xad::NoTapeException);
```

Comment:
The `const` version gets called when the method is executed on a `const` instance of the variable.

```suggestion
    const xad::AD y = 1.0;
    EXPECT_THROW(y.derivative(), xad::NoTapeException);
```

Reply:
Thanks, I missed that!